content/views.py

```diff
@@ -43,8 +69,12 @@ def articles(self, request, pk=None):
 class ArticleViewSet(ModelViewSet):
     queryset = Article.objects.all()
     serializer_class = ArticleSerializer
-
     permission_classes = [IsAuthenticatedOrReadOnly, IsArticleOwnerOrReadOnly]
+    filter_backends = [SearchableFilterBackend]
+    search_fields = ["title", "content_raw"]
```

Comment:
So I think you'll agree with me that we shouldn't be doing full text searches on the raw JSON article format. Wanted to get your thoughts on how exactly to handle this? My one idea so far is a plain text version that we update periodically and search on instead.

Reply:
A plain text version seems like it might waste space, but I agree, searching the raw JSON would be a pain.

I found two papers for efficient searching within compressed text: https://arxiv.org/pdf/cs/0505056.pdf, http://corpus.canterbury.ac.nz/research/search_bwt.pdf. If there was a way to get SQLite/MariaDB to use these algorithms, it would make life much easier. Until then, your idea is probably the best.

Reply:
Ok, so on second thought we could compress the actual `raw_content` field, and store plaintext as suggested. The upside is that since it's JSON and there's likely to be a lot of repetition, we can gain a lot of savings. The downside is that for good performance with automatic saving and live collaboration in the future, we may need to save to an in-memory database like redis and periodically commit changes to the backend.

Reply:
I'd be interested in knowing how much disk space we're taking up right now with the old site. If it's like 100 MB, I think we can get away with anything.